static/app/utils/number/ladder.tsx

```diff
@@ -0,0 +1,48 @@
+import orderBy from 'lodash/orderBy';
+
+type Rung<T> = [threshold: number, value: T];
+
+/**
+ * Class that represents a value ladder. Given a ladder of increasing thresholds, the ladder can match an incoming value against the known intervals.
```

Comment:
Any chance we could add some examples and maybe an example use case?

Reply:
Yes! Added